include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -1078,29 +1080,107 @@ void RunInParallelSection(ThreadPoolParallelSection &ps,
 // Run a single parallel loop _without_ a parallel section.  This is a
 // special case of RunInParallelSection, avoiding code paths for
 // handing off multiple loops to the pool of workers.
-
 void RunInParallel(std::function<void(unsigned idx)> fn, unsigned n, std::ptrdiff_t block_size) override {
   profiler_.LogStartAndCoreAndBlock(block_size);
-  PerThread *pt = GetPerThread();
+  PerThread* pt = GetPerThread();
   ThreadPoolParallelSection ps;
   StartParallelSectionInternal(*pt, ps);
 
-  // Summon workers to run the function (n is the desired maximum
-  // degree of parallelism, including the main thread).  Unlike the
-  // multi-loop RunInParallelSection, this single-loop worker can run
-  // fn directly without needing to receive it via ps.current_loop.
-  SummonWorkers(*pt, ps, n, fn);
-  profiler_.LogEndAndStart(ThreadPoolProfiler::DISTRIBUTION);
+  // index of thread that dispatch work to all other threads
+  int dispatch_q_idx = -1;
+  int dispatch_w_idx = -1;  // index of enqueued work
+  std::atomic<bool> dispatch_done{false};
+  std::atomic<bool> dispatch_work_done{false};
 
-  // Run work in the main thread
-  fn(0);
+  Task worker_fn = [&]() {
```

Comment:
Can we expand the [&] to be explicit about what is being captured.  A common bug is to implicitly capture local variables, and then access them once they are out of scope.  Making the captures explicit makes it easier to check that we are only handling safe cases (e.g., by-value capture from this scope, and capture of reference-typed variables).

Reply:
Done.